Unable to build individual components with "./mach cargo check -p"

`servo_url` gives errors about `Serialize`:
```
error: cannot find derive macro `Serialize` in this scope
  --> components/url/origin.rs:13:66
   |
13 | #[derive(Clone, Debug, Deserialize, Eq, MallocSizeOf, PartialEq, Serialize)]
   |                                                                  ^^^^^^^^^

error: cannot find derive macro `Deserialize` in this scope
  --> components/url/origin.rs:13:24
   |
13 | #[derive(Clone, Debug, Deserialize, Eq, MallocSizeOf, PartialEq, Serialize)]
   |                        ^^^^^^^^^^^

error: cannot find derive macro `Serialize` in this scope
  --> components/url/origin.rs:97:52
   |
97 | #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
   |                                                    ^^^^^^^^^

error: cannot find derive macro `Deserialize` in this scope
  --> components/url/origin.rs:97:24
   |
97 | #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
   |                        ^^^^^^^^^^^

error[E0658]: The attribute `serde` is currently unknown to the compiler and may have meaning added to it in the future (see issue #29642)
  --> components/url/origin.rs:21:9
   |
21 |         #[serde(deserialize_with = "url_serde::deserialize", serialize_with = "url_serde::serialize")]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(custom_attribute)] to the crate attributes to enable

error: aborting due to 5 previous errors
```

`compositing` is complaining about `MallocSizeOf`:
```
error[E0277]: the trait bound `url::Host: malloc_size_of::MallocSizeOf` is not satisfied
  --> components/url/origin.rs:13:41
   |
13 | #[derive(Clone, Debug, Deserialize, Eq, MallocSizeOf, PartialEq, Serialize)]
   |                                         ^^^^^^^^^^^^ the trait `malloc_size_of::MallocSizeOf` is not implemented for `url::Host`
   |
   = note: required by `malloc_size_of::MallocSizeOf::size_of`
```

--- a/components/url/Cargo.toml
+++ b/components/url/Cargo.toml
@@ -10,9 +10,9 @@ name = "servo_url"
 path = "lib.rs"
 
 [dependencies]
-malloc_size_of = { path = "../malloc_size_of" }
+malloc_size_of = { path = "../malloc_size_of", features = ["servo"] }
 malloc_size_of_derive = { path = "../malloc_size_of_derive" }
-serde = {version = "1.0"}
+serde = {version = "1.0", features = ["derive"]}
 servo_rand = {path = "../rand"}
 url = "1.2"
 url_serde = {version = "0.2"}

